crates/solver/src/settlement_rater.rs

```diff
@@ -322,46 +352,48 @@ impl From<ScoringError> for anyhow::Error {
 
 /// Contains a subset of the configuration options for the submission of a
 /// settlement, needed for the score calculation.
-pub struct SubmissionConfig {
-    pub strategies: Vec<TransactionStrategyArg>,
+#[derive(Debug, Clone)]
+pub struct SubmissionStrategy {
+    pub strategy: TransactionStrategyArg,
```

Comment:
nit: since `disable_high_risk_public_mempool_transactions` is global for the entire process it should be on `ScoreCalculator` and not on each strategy.

Reply:
It felt nice to me to have the config per strategy, as it fits nice with the colocated driver config for mempools, but I agree your comment makes sense.
Good you mentioned this, because I noticed a bug https://github.com/cowprotocol/services/pull/1856/commits/3921f2a7649051a3d3bf665b580e9a089c2df7a6

Reply:
This comment is still outstanding

Reply:
https://github.com/cowprotocol/services/pull/1856/commits/8fd364e98b058294eb03d1fcb431381f817d312c